backend/app/services/index/ElasticsearchResources.scala

```diff
@@ -442,13 +442,18 @@ class ElasticsearchResources(override val client: ElasticClient, indexName: Stri
 
   }
 
-  def getBlobs(collection: String, ingestion: String, size: Int): Attempt[Iterable[IndexedBlob]] = {
+  def getBlobs(collection: String, maybeIngestion: Option[String], size: Int): Attempt[Iterable[IndexedBlob]] = {
+    val query = maybeIngestion match {
+      case Some(ingestion) => matchQuery(IndexFields.ingestionRaw, s"$collection/$ingestion")
+      case _ => matchQuery(IndexFields.collectionRaw, collection)
+    }
+
```

Comment:
scala people always used to tell me not to pattern match on an option, an alternative would be:

```
val path = ingestion.map(i => "$collection/$i).getOrElse(collection)
val query = matchQuery(IndexFields.collectionRaw, path`
```

Reply:
oops sorry, it's funny I did think I'd looked at the first argument but clearly not. 

here's the argument: https://www.originate.com/thinking/idiomatic-scala-your-options-do-not-match

I think pattern matching is probably more readable than the map getOrElse pattern though, certainly more readable than `fold`

Reply:
Interesting. Though I totally reject their premise that concise = better. Why do we even bother naming our variables? It would be more concise to just call them `a`, `b`, etc.

The type safety point is the most compelling. I hadn't spotted that in my pattern match the type had become `Any` because one of the matches returned `Attempt[Ingestion]` and the other `Attempt[Collection]`